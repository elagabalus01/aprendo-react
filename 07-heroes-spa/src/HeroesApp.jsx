import {AppRouter} from "./router/AppRouter.jsx";

export const HeroesApp = ()=>{
    return (
        <>
            <h1>Heroes app</h1>
            <AppRouter/>

        </>
    )
}
