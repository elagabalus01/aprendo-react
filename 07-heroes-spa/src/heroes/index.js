export * from "./pages/index.js";
export * from "./routes/HeroesRoutes.jsx";
