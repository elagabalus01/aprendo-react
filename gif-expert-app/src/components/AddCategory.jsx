import {useState} from "react";
import {findRenderedDOMComponentWithTag} from "react-dom/test-utils";
import PropTypes from "prop-types";

export const AddCategory = ({onNewCategory})=> {
    const [inputValue, setInputValue] = useState('')
    const onInputChange = ({target})=>{
        console.log(target)
        setInputValue(target.value)
    }


    const onSubmit = (event)=>{
        event.preventDefault()
        console.log(inputValue)
        if(inputValue.trim().length <= 1){return}
        /*setCategories((currentValue)=>{
            return [...currentValue, inputValue]
        })*/
        onNewCategory(inputValue.trim())
        setInputValue('')
    }

    return (
        <form onSubmit={onSubmit} aria-label={"form"}>
            <input
                type="text"
                placeholder="Buscar gif"
                value = { inputValue}
                onChange={onInputChange}
            />
        </form>
    )

}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}
