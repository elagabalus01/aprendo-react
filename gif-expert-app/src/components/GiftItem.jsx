import PropTypes from "prop-types";


const GiftItem = ({title, url})=>{

    return (
        <>
        <div className="card">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>


        </>
    )
}

GiftItem.prototype = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

export {
    GiftItem
};

