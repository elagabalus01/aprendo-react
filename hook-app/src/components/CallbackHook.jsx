import {useCallback, useState} from "react";
import {ShowIncremente} from "./ShowIncremente.jsx";

export const CallbackHook = ()=>{
    const [counter, setCounter] = useState(10)

    const incrementeFather = useCallback(
        (value = 0) => {
        setCounter((counter)=>counter + value)
    }, []);


    //const incrementeFather = ()=>{
    //    setCounter(counter + 1)
    //}


    return (
        <>
            <h1>
                useCallback hook: {counter}
            </h1>
            <ShowIncremente incrementeFunction={incrementeFather}></ShowIncremente>
        </>
    )
}
