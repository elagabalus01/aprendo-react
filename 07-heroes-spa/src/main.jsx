import React from 'react'
import ReactDOM from 'react-dom/client'
import {HeroesApp} from "./HeroesApp.jsx";
import {BrowserRouter} from "react-router-dom";
import {AuthProvider} from "./auth/index.js";

ReactDOM.createRoot(document.getElementById('root')).render(
    //<React.StrictMode>
    <BrowserRouter>
        <AuthProvider>
            <HeroesApp/>
        </AuthProvider>
    </BrowserRouter>

    //</React.StrictMode>,
)
