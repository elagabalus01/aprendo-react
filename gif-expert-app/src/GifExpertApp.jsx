import {useState} from "react";
import {AddCategory, GifGrid} from "./components";
export const GifExpertApp = ()=>{

    const [categories, setCategories]= useState([])

    const onAddCategory = (category)=>{
        console.log("AGREGANDO")
        if(categories.includes(category)) return;
        setCategories((currentCategories)=>{
            return [...currentCategories, category]
        })
    }

    return (<>
        <h1>GIF EXPERT APP</h1>
        <AddCategory onNewCategory={(event)=>onAddCategory(event)}></AddCategory>
        {categories.map(
            categoria=> <GifGrid key={categoria} category={categoria}></GifGrid>
        )}

        <button onClick={onAddCategory}>Agregar</button>

        </>)
}
