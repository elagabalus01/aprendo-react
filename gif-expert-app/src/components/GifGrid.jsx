import {getGifts} from '../helpers/getGIfts.js'
import {useEffect, useState} from "react";
import {GiftItem} from "./GiftItem.jsx";
import '../css/styles.css'
export const GifGrid = ({category})=>{

    const [images, setImages] = useState([])

    const [counter, setCounter] = useState(10)

    useEffect(() => {
        getGifts(category).then((images)=>
            setImages(images)
        )
    }, []);



    return (
        <>
            <h3>{category}</h3>
            <div className="card-grid">
                {
                    images.map(({id, title, url})=>(
                    <GiftItem key={id} title={title} url={url} />
                ))
            }
            </div>
            <p>{counter}</p>
            <button onClick={()=> setCounter(counter + 1)} > +1</button>
        </>
    )
}
