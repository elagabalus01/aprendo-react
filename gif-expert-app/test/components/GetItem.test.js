import {render, screen} from "@testing-library/react";
import {GiftItem} from "../../src/components";


describe("Prueba con propetypes", ()=>{
    const title = "titulo desde prueba"
    const url = "www.google.com"

    test("Prueba", ()=>{
        console.log("Hola mudno")

        const {container, getByText} = render(<GiftItem title={title} url={url}/>)
        ///expect(container).toMatchSnapshot()
        expect(getByText(title)).toBeTruthy()

        expect(container).toMatchSnapshot()

    })

    test("Prueba", ()=>{
        /*expect(render(<GiftItem/>)).toThrow();*/
    })

    test("Asignacion de propiedades", ()=>{
        render(<GiftItem title={title} url={url}/>)
        const imgTag = screen.getByRole("img")
        expect(imgTag.src).toContain(url)
        expect(imgTag.alt).toBe(title)
    })


    test("Debe contener el titulo", ()=>{
        render(<GiftItem title={title} url={url}/>)
        expect(screen.getByText(title)).toBeTruthy()
    })

})
