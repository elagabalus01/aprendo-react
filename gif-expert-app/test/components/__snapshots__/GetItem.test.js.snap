// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba con propetypes Prueba 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="titulo desde prueba"
      src="www.google.com"
    />
    <p>
      titulo desde prueba
    </p>
  </div>
</div>
`;
